### ISSUE TITLE
`undefined` errors after adding `com_google_cloud_go_compute` dependency

### ISSUE THREAD

**Issue by amalucelli** [bazelbuild/bazel-gazelle#1159]
### What version of gazelle are you using?

`v0.24.0`

### What version of rules_go are you using?

`v0.29.0`

### What version of Bazel are you using?

`4.2.1`

### Does this issue reproduce with the latest releases of all the above?

Yes

### What operating system and processor architecture are you using?

- `linux/amd64`
- `darwin/adm64`

### What did you do?

Tried to add a new dependency for `cloud.google.com/go/compute` version `v1.0.0` and getting a lot of `undefined` errors for that dependency in the build/test stage.

Here is the `go_repository` for that dependency:

```go
go_repository(
    name = "com_google_cloud_go_compute",
    importpath = "cloud.google.com/go/compute",
    sum = "h1:SJYBzih8Jj9EUm6IDirxKG0I0AGWduhtb6BmdqWarw4=",
    version = "v1.0.0",
)
```

And this is the Go version I'm using:

```go
go_register_toolchains(version = "1.17.4")
```

I even saw some issue with this dependency in #1154, but I think is unrelated. I even tried the workaround provided, changing the order but no success.

### What did you expect to see?

No build errors.

### What did you see instead?

```go
external/com_google_cloud_go_compute/apiv1/image_family_views_client.go:49:24: undefined: compute.GetImageFamilyViewRequest
external/com_google_cloud_go_compute/apiv1/image_family_views_client.go:49:82: undefined: compute.ImageFamilyView
external/com_google_cloud_go_compute/apiv1/image_family_views_client.go:87:64: undefined: compute.GetImageFamilyViewRequest
external/com_google_cloud_go_compute/apiv1/image_family_views_client.go:87:127: undefined: compute.ImageFamilyView
external/com_google_cloud_go_compute/apiv1/image_family_views_client.go:156:68: undefined: compute.GetImageFamilyViewRequest
external/com_google_cloud_go_compute/apiv1/image_family_views_client.go:156:131: undefined: compute.ImageFamilyView
external/com_google_cloud_go_compute/apiv1/instances_client.go:116:44: undefined: compute.SendDiagnosticInterruptInstanceRequest
external/com_google_cloud_go_compute/apiv1/instances_client.go:116:115: undefined: compute.SendDiagnosticInterruptInstanceResponse
external/com_google_cloud_go_compute/apiv1/instances_client.go:276:77: undefined: compute.SendDiagnosticInterruptInstanceRequest
external/com_google_cloud_go_compute/apiv1/instances_client.go:276:153: undefined: compute.SendDiagnosticInterruptInstanceResponse
external/com_google_cloud_go_compute/apiv1/instances_client.go:276:153: too many errors
```

**Comment by amalucelli** [comment]
👋🏻 @blico, thanks for your help, I actually tried with that dependency `v0.0.0-20211221195035-429b39de9b1c`, but still got the same error.

The one I'm currently using is this one `v0.0.0-20220111164026-67b88f271998` (_a newer_):

```go
go_repository(
    name = "org_golang_google_genproto",
    importpath = "google.golang.org/genproto",
    sum = "h1:g/x+MYjJYDEP3OBCYYmwIbt4x6k3gryb+ohyOR7PXfI=",
    version = "v0.0.0-20220111164026-67b88f271998",
)
```

So apparently I have everything right 🤔 I even reviewed all the other dependencies declared in that `go.mod` and everything is up-to-date.

Recently I had to change the order of the dependencies as I described in this ticket/comment https://github.com/bazelbuild/bazel-gazelle/issues/1140#issuecomment-988043980, so it seems to be in the right order also.
